scripts/check_cpp_format.py

```diff
@@ -0,0 +1,133 @@
+#!/usr/bin/python
+
+import fnmatch
+import io
+import multiprocessing
+import os
+import queue
+import subprocess
+import sys
+import threading
+
+# Parse command-line arguments
+if len(sys.argv) == 1:
+    print("Usage: check_cpp_format.py [-fix] <paths>")
+    sys.exit()
+i_arg = 1
+fix = False
+if sys.argv[i_arg] == "-fix":
+    fix = True
+    i_arg += 1
+roots = []
+for root in sys.argv[i_arg:]:
+    roots.append(root)
+
+# Parse environment variables
+prune_paths = set()
+if "PRUNE_PATHS" in os.environ:
+    last_token = io.StringIO()
+    in_quotes = False
+    for ch in os.environ["PRUNE_PATHS"]:
+        if ch in ["'", '"']:
+            if last_token.tell() != 0:
+                prune_paths.add(last_token.getvalue())
+                last_token = io.StringIO()
+            in_quotes = not in_quotes
+        elif ch == " ":
+            if in_quotes:
+                last_token.write(ch)
+            elif last_token.tell() != 0:
+                prune_paths.add(last_token.getvalue())
+                last_token = io.StringIO()
+            # else tokens are separated by multiple spaces
+        else:
+            last_token.write(ch)
+    if last_token.tell() != 0:
+        prune_paths.add(last_token.getvalue())
+        last_token = io.StringIO()
+clang_format = os.environ.get("CLANG_FORMAT", "clang-format")
+
+# Hard-coded constants
+prune_names = {"build*", ".#*"}
+file_suffixes = {".cpp", ".h", ".cu", ".cuh"}
+
+
+# Create the task queue
+tasks = queue.Queue()
+
+# Launch consumer threads
+class Consumer:
+    def __init__(self):
+        self.thread = threading.Thread(target=self.entry)
+        self.check_failed = False
```

Comment:
Subclass `Thread`.

Reply:
I don't want all the `Thread` members to leak into `Consumer`. Composition is better than inheritance in this case. Because I don't need e.g. to call `Consumer` methods by a reference to `Thread` object, don't need to override the methods, etc.